nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -193,4 +205,80 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
         notifyEvent(session, eventType, commentParent, commentDocumentModel);
     }
 
+    /**
+     * Checks if a document has comments.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected boolean hasComments(CoreSession session, DocumentModel document) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Checks if a document has comments from a particular user.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected boolean hasCommentsByUser(CoreSession session, DocumentModel document, String author) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Subscribes top level document's author on first comment and commentor on his first comment if autosubscribtion is
+     * turned on.
```

Comment:
i think we should rework the javadoc to be more readable like `Subscribes the comment author on events to the top level document.` for more context on for example which event are added during this subscription you can use  
`@implspec`

Reply:
```/**
     * Handles autosubscription to COMMENT_ADDED_NOTIFICATION following notification rules.
     *
     * @implSpec The default implementation autosubscribes top level document's author on first comment if
     *           autosubscription is enabled. This relies on the implementation of the abstract hasComments method
     * @implSpec The default implementation autosubscribes comment author on his first comment on top level document if
     *           autosubscription is enabled. This relies on the implementation of the abstract hasCommentsByUser method
     * @param session the core session
     * @param commentDocModel the comment
     * @param document the document being commented
     * @since 11.1
     */
```

Reply:
Extra feedbacks are welcome